hack/make-rules/test.sh

```diff
@@ -384,7 +384,7 @@ for (( i=0; i<${apiVersionsCount}; i++ )); do
   apiVersion=${apiVersions[i]}
   echo "Running tests for APIVersion: $apiVersion"
   # KUBE_TEST_API sets the version of each group to be tested.
-  KUBE_TEST_API="${apiVersion}" runTests "$@"
+  KUBE_TEST_API="${apiVersion}" runTests "$@" && RC=$? || RC=$?
```

Comment:
`; RC=$?`

Reply:
hehe, this was a great bash exercise, I was wondering why the [ternary operator a && b || c was used in other parts of the code](https://github.com/aojea/kubernetes/blob/76e160b815109eb5e8294352a800804d00e9658b/hack/make-rules/test.sh#L286)  and you recommend use a semicolon here. 
My conclusion is that you need  && (and also ||) to handle a single exit code for all the chained commands.

Reply:
in our scripts we fail hard on non-zero error codes. I.e. `false; RC=$?` never reaches the assignments. In contrast, `false && RC=$? || RC=$?` executes `false`, then skips `RC=$?`, then calls `RC=$?`. The whole command will have a zero return code, while RC will be the return code of `false`. So, my suggestion was probably wrong in this context.

Reply:
will revert then, also that formula is used in other parts of the code https://github.com/aojea/kubernetes/blob/a1cd0b9be21e6d9703f1ab2aef1e9ca56576a931/hack/make-rules/test.sh#L286

Reply:
Yeah, usually you use that && || construction if you want to store the return code in RC, but then basically ignore it (so that the bash will not quit your script).

Reply:
Looking at the code: where would `RC` be processed after your change in the first commit?

Reply:
it doesn't, it only catch the error to allow that the script ends and avoid the stack trace reported in #60946

Reply:
Only seeing `reportCoverageToCoveralls` below this line. Somehow RC must be processed. With the current code a test error is ignored, isn't it?

Reply:
basically you are changing a test command `go test` to `go test || true` which ignores the error. Doesn't look like what we want either, does it?

Reply:
I think that's the question, why do we return the resulting error if it's not processed? 
do we want to fail the script if one of the tests fail?  if affirmative the stack trace is the expected behavior, isn't it?

Reply:
`git blame` points to @thockin , can you help us to understand why the error code is returned in the `runTests` function?

Reply:
I would expect the script to fail when tests fail. Isn't that the usual test script logic?